### PR TITLE
Latest changes on the annotation service

### PR DESCRIPTION
**Pull request by tanksha**
The latest changes to the annotation service includes:
- No global variables are used
- Localization info added for Genes and Molecule Nodes for the cell visualizer
- Scheme output modified to use ListLink instead of scheme list
- Function names with underscore are replaced by '-' to meet the standard
- The refactored Gene-Go annotation code is merged
- The scheme result is tested with current scheme2Csv converter, only modifying the parser is needed.
- The requests from the front-end needs to be modified to meet the current changes.

### CODE REVIEW

File: functions/annotation_request_handler.scm [hunk 1]
```diff
@@ -1,45 +1,42 @@
...
 (for-each (lambda(g)
-    (if (equal? (cog-node 'GeneNode g) '())  ;; validates if given gene symbols exist in the atomspace
+    (if (equal? (cog-node 'GeneNode g) '())  
       (set! unknown (append unknown (list g)) )
-      (set! gene_nodes (append gene_nodes (list (GeneNode g))))
     )
-  ) (string-split gene_list #\ ))
+  ) gene-list)
 
 (if (equal? unknown '())
```
Comment (Habush) [inline functions/annotation_request_handler.scm]: This can be simplified with `(null? unknown)`

File: functions/pm_functions.scm [hunk 2]
```diff
@@ -435,13 +457,111 @@
...
+		    (VariableNode "$b")
+		  ))
 	
-	))
+))
+))
+
+;; Grounded schema node to add info about matched variable nodes
+
+(define (generate-result var1 var2)
```
Comment (Habush) [inline functions/pm_functions.scm]: Should this function be in this file as it isn't doing any pattern matching? Better if it is moved to `utils.scm`

File: functions/pm_functions.scm [hunk 3]
```diff
@@ -435,13 +457,111 @@
...
+          (node-info var2)
+          (node-info var1)
+          )])
+    output
+    )
+    )
+))
+;; build description URL of a node
+
+(define (build-desc-url node)
```
Comment (Habush) [inline functions/pm_functions.scm]: Should this function be in this file as it isn't doing any pattern matching? Better if it is moved to `utils.scm`

**Comment by tanksha** [comment]
@Habush I have moved the suggested non PM functions to utils. @enku-io If you don't have any comments then, it can be merged now I think.

**Pull request merged**